src/index.ts

```diff
@@ -62,3 +58,13 @@ export function resolve(
 
   return result
 }
+
+export function createCustomResolver(options: ResolverOptions) {
+  return {
+    name: 'eslint-import-resolver-x',
+    interfaceVersion: 3,
+    resolve(modulePath: string, source: string) {
+      return resolve(modulePath, source, options)
+    },
```

Comment:
This does work, but you may be able to hoist some logic before `return` (e.g. instead of a keyed-cache, you can now use singleton).

Reply:
@SukkaW fixed